cmd/relay/bgs/bgs.go

```diff
@@ -0,0 +1,1376 @@
+			return fmt.Errorf("failed to un-tombstone a user: %w", err)
+		}
+		u.SetTombstoned(false)
+
+		//ai, err := bgs.Index.LookupUser(ctx, u.ID)
+		//if err != nil {
+		//	repoCommitsResultCounter.WithLabelValues(host.Host, "nou2").Inc()
+		//	return fmt.Errorf("failed to look up user (tombstone recover): %w", err)
+		//}
+
+		// Now a simple re-crawl should suffice to bring the user back online
+		//repoCommitsResultCounter.WithLabelValues(host.Host, "catchupt").Inc()
+		//return bgs.Index.Crawler.AddToCatchupQueue(ctx, host, ai, evt)
+		// TODO: fall through and just handle the event and the right thing should happen? -- bolson 2025 unsure
+	}
+
+	var prevState UserPreviousState
+	err = bgs.db.First(&prevState, u.ID).Error
+	//prevP := &prevState
+	var prevP repomgr.UserPrev = &prevState
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		prevP = nil
+	} else if err != nil {
+		bgs.log.Error("failed to get previous root", "err", err)
+		prevP = nil
+	}
+	dbPrevRootStr := ""
+	dbPrevSeqStr := ""
+	if prevP != nil {
+		if prevState.Seq >= evt.Seq && ((prevState.Seq - evt.Seq) < 2000) {
+			// ignore catchup overlap of 200 on some subscribeRepos restarts
+			repoCommitsResultCounter.WithLabelValues(host.Host, "dup").Inc()
+			return nil
+		}
+		dbPrevRootStr = prevState.Cid.CID.String()
+		dbPrevSeqStr = strconv.FormatInt(prevState.Seq, 10)
+	}
+	evtPrevDataStr := ""
+	if evt.PrevData != nil {
+		evtPrevDataStr = ((*cid.Cid)(evt.PrevData)).String()
+	}
+	newRootCid, err := bgs.repoman.HandleCommit(ctx, host, u, evt, prevP)
+	if err != nil {
+		bgs.inductionTraceLog.Error("commit bad", "seq", evt.Seq, "pseq", dbPrevSeqStr, "pdsHost", host.Host, "repo", evt.Repo, "prev", evtPrevDataStr, "dbprev", dbPrevRootStr, "err", err)
+		bgs.log.Warn("failed handling event", "err", err, "pdsHost", host.Host, "seq", evt.Seq, "repo", u.Did, "commit", evt.Commit.String())
+		repoCommitsResultCounter.WithLabelValues(host.Host, "err").Inc()
+		return fmt.Errorf("handle user event failed: %w", err)
+	} else {
+		// store now verified new repo state
+		prevState.Uid = u.ID
+		prevState.Cid.CID = *newRootCid
+		prevState.Rev = evt.Rev
+		prevState.Seq = evt.Seq
+		bgs.inductionTraceLog.Info("commit  ok", "seq", evt.Seq, "pdsHost", host.Host, "repo", evt.Repo, "prev", evtPrevDataStr, "dbprev", dbPrevRootStr, "nextprev", newRootCid.String())
+		if prevP == nil {
+			err = bgs.db.Create(&prevState).Error
+			if err != nil {
+				return fmt.Errorf("failed to create previous root uid=%d: %w", u.ID, err)
+			}
+		} else {
+			err = bgs.db.Save(&prevState).Error
+			if err != nil {
+				return fmt.Errorf("failed to save previous root uid=%d: %w", u.ID, err)
+			}
+		}
+	}
+
+	repoCommitsResultCounter.WithLabelValues(host.Host, "ok").Inc()
+	return nil
+}
+
+func (bgs *BGS) handleSync(ctx context.Context, host *models.PDS, evt *comatproto.SyncSubscribeRepos_Sync) error {
+	// TODO: actually do something with #sync event
```

Comment:
we should at least be updating the metadata for the account (DID) here, right?

Reply:
I think the rough expected behavior is:

- parse out the commit object
- verify the message matches the fields in the commit object
- fetch user metadata
    - if account isn't active, drop event
    - if rev is old/bad, drop event
- update user metadata with sync info
- emit the event